packages/strapi-plugin-users-permissions/config/schema.graphql.js

```diff
@@ -182,7 +182,14 @@ module.exports = {
         description: 'Register a user',
         resolverOf: 'plugins::users-permissions.auth.register',
         resolver: async (obj, options, { context }) => {
-          context.request.body = _.toPlainObject(options.input);
+          const input = _.toPlainObject(options.input);
+
+          context.request.body = _.omit(input, [
+            'resetPasswordToken',
+            'confirmed',
+            'blocked',
+            'role',
```

Comment:
Not sure if role is considered private

Reply:
Oh it is, the user shouldn't be responsible for choosing his own role. In fact Strapi already [overwrites it](https://github.com/gfpacheco/strapi/blob/master/packages/strapi-plugin-users-permissions/controllers/Auth.js#L464) anyway.

I wanted to remove the fields from the GraphQL input schema, but it's using UserInput so I'm just ignoring them even if the user sends it.